packages/vitest-angular/src/lib/builders/build/vitest.impl.ts

```diff
@@ -0,0 +1,216 @@
+    setupFiles: [setupFile],
+    globals: true,
+    pool: 'vmThreads',
+    reporters: ['default'],
+    environment: 'jsdom',
+    exclude: options?.exclude || [],
+    ...extraArgs,
+  };
+
+  const includes: string[] = findIncludes({
+    workspaceRoot,
+    projectRoot,
+    include: options.include,
+    exclude: options.exclude || [],
+  });
+
+  const testFiles = [
+    path.relative(workspaceRoot, options.setupFile),
+    ...includes.map((inc) => path.relative(workspaceRoot, inc)),
+  ];
+
+  const entryPoints = generateEntryPoints({
+    projectRoot: projectRoot,
+    testFiles,
+    context,
+    angularVersion,
+  });
+
+  const outputFiles = new Map();
+
+  const viteConfig: UserConfig = {
+    plugins: [
+      (await createAngularMemoryPlugin({
+        angularVersion,
+        workspaceRoot,
+        outputFiles,
+      })) as Plugin,
+      await esbuildDownlevelPlugin(),
+    ],
+  };
+
+  let server: Vitest | undefined;
+  for await (const buildOutput of buildApplicationInternal(
+    {
+      aot: false,
+      index: false,
+      progress: false,
+      prerender: false,
+      optimization: false,
+      outputPath: `.angular/.vitest/${projectConfig['name']}`,
+      tsConfig: path.relative(workspaceRoot, options.tsConfig),
+      watch: options.watch === true,
+      entryPoints,
+      allowedCommonJsDependencies: ['@analogjs/vitest-angular/setup-zone'],
+    },
+    context
+  )) {
+    if (buildOutput.kind === ResultKind.Failure) {
+      yield { success: false };
+    } else if (
+      buildOutput.kind === ResultKind.Incremental ||
+      buildOutput.kind === ResultKind.Full
+    ) {
+      if (buildOutput.kind === ResultKind.Full) {
+        outputFiles.clear();
+        Object.keys(buildOutput.files).forEach((key) => {
+          outputFiles.set(key, buildOutput.files[key]);
+        });
+      } else {
+        Object.keys(buildOutput.files).forEach((key) => {
+          outputFiles.set(key, buildOutput.files[key]);
+        });
+      }
+    }
+
+    if (options.watch) {
+      const vitestServer = await startVitest('test', [], config, viteConfig);
+      server = vitestServer;
+
+      yield { success: true };
+    } else {
+      server = await startVitest('test', [], config, viteConfig);
+
+      const success = server?.state.getCountOfFailedTests() === 0;
+
+      yield { success };
+    }
+  }
+
+  yield { success: true };
+}
+
+export async function getExtraArgs(
+  options: VitestSchema
+): Promise<Record<string, any>> {
+  // support passing extra args to Vitest CLI
+  const schema = await import('./schema.json');
+  const extraArgs: Record<string, any> = {};
+  for (const key of Object.keys(options)) {
+    if (!(schema as any).properties[key]) {
+      extraArgs[key] = (options as any)[key];
+    }
+  }
+
+  return extraArgs;
+}
+
+function findIncludes(options: {
+  workspaceRoot: string;
+  projectRoot: string;
+  include: string[];
+  exclude: string[];
+}) {
+  const fg = require('fast-glob');
```

Comment:
What do you think about some "better" alternative https://github.com/es-tooling/module-replacements/blob/main/docs/modules/glob.md ([e18e](https://e18e.dev/)).

Reply:
Yea, I don't have strong feelings about switching. Happy to discuss in an issue though